### ISSUE TITLE
Any plans for PHP7?

### ISSUE THREAD

**Issue by akademic** [alexeyrybak/blitz#24]
Trying to test my project on PHP7, but Blitz does not compile.

**Comment by alexeyrybak** [comment]
Blitz will run with PHP7 when PHP7 will be more or less close to the production-ready state.

**Comment by alexeyrybak** [comment]
OK, few more details on this. We'll start fixing all the extensions when PHP7 becomes at least RC1. So if this happens say in July-October as they promise - we'll start working on a PHP7 version.

**Comment by akademic** [comment]
Sounds great!

**Comment by Mrhyuan** [comment]
Great! RC 1 will be released on the 20th of August.

**Comment by akademic** [comment]
RC4 is out. Any news?

**Comment by alexeyrybak** [comment]
We are working on it. See tony's branch https://github.com/tony2001/blitz/commit/a4290126c7a7facb5f51dfe39c706f3a23ecf877

**Comment by akademic** [comment]
Good news! :)

**Comment by alexeyrybak** [comment]
Same in my repo: https://github.com/alexeyrybak/blitz/tree/php7. Most of test passed, but not tested at prod yet.

**Comment by denji** [comment]
https://github.com/alexeyrybak/blitz/tree/php7

```
$ cppcheck -f --enable=warning,performance,portability -j4 . 2>
blitz_php7-err.txt
[blitz.c:1250]: (error) Converting pointer arithmetic result to bool. The
bool is always true unless there is undefined behaviour.
[blitz.c:2961]: (warning) %u in format string (no. 1) requires 'unsigned
int' but the argument type is 'int'.
[blitz.c:3418]: (warning) %u in format string (no. 1) requires 'unsigned
int' but the argument type is 'long'.
[blitz.c:4235] -> [blitz.c:5368]: (warning) Possible null pointer
dereference: local - otherwise it is redundant to check it against null.
[blitz.c:1744] -> [blitz.c:1745]: (performance) Variable 'i_pos' is
reassigned a value before the old one has been used.
[blitz.c:1763] -> [blitz.c:1764]: (performance) Variable 'i_pos' is
reassigned a value before the old one has been used.
[blitz.c:3943] -> [blitz.c:3948]: (performance) Variable 'node' is
reassigned a value before the old one has been used.
[blitz.c:5585] -> [blitz.c:5593]: (performance) Variable 'res' is
reassigned a value before the old one has been used.
[blitz.c:5691] -> [blitz.c:5696]: (performance) Variable 'res' is
reassigned a value before the old one has been used.
[blitz.c:4008]: (warning) Logical disjunction always evaluates to true:
EXPR != 106 || EXPR != 114.
```

2015-10-08 12:20 GMT+03:00 Alexey Rybak <notifications@github.com>:

> Same in my repo: https://github.com/alexeyrybak/blitz/tree/php7. Most of
> test passed, but not tested at prod yet.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/alexeyrybak/blitz/issues/24#issuecomment-146468134>.
>



-- 
Sincerely,
Denis Denisov

**Comment by shumeev** [comment]
/usr/local/php7/bin/phpize - ok
./configure --with-php-config=/usr/local/php7/bin/php-config - ok
make - error
![screenshot_1](https://cloud.githubusercontent.com/assets/9616836/10413795/1baf053e-6fc8-11e5-8eec-3ad177c8ca1f.png)
find / -name php_smart_str.h - not found !

What could be the problem?

**Comment by akademic** [comment]
I have blitz-0.8.2 at my production.
Today launched my project on PHP7 (from dotdeb.org) with new blitz (from https://github.com/alexeyrybak/blitz/tree/php7)
All works as expected.

**Comment by alexeyrybak** [comment]
Cool, thanks. We have two tests broken, but this seems to be minor. I will dig into that a bit later, and we'll make a release.

**Comment by alexeyrybak** [comment]
All tests fixed (there was one bug), see 0.10.1 (PHP7) release.
